src/toil_scripts/adam_pipeline/adam_preprocessing.py

```diff
@@ -354,66 +353,67 @@ def scale_external_spark_cluster(num_samples=1):
-                        help = 'Name of the jobstore')
-    parser.add_argument('-m', '--master_ip', required = False,
-                        help="IP or hostname of host running for Spark master and HDFS namenode. Should be provided "
-                             "if pointing at a static (external or standalone) Spark cluster. The special value "
-                             "'auto' indicates the master of standalone cluster, i.e. one that is managed by the "
-                             "uberscript.")
-    parser.add_argument('-u', '--sudo',
-                        dest='sudo', action='store_true',
-                        help='Docker usually needs sudo to execute '
-                        'locally, but not''when running Mesos '
-                        'or when a member of a Docker group.')
-
-    return parser
-
-# FIXME: unused parameter args
-
-def main(args):
-
-    parser = build_parser()
-    Job.Runner.addToilOptions(parser)
-    options = parser.parse_args()
-
-    if bool(options.master_ip) == bool(options.num_nodes):
-        raise ValueError("Only one of --master_ip (%s) and --num_nodes (%d) can be provided." %
-                         (options.master_ip, options.num_nodes))
-
-    if options.num_nodes <= 1:
-        raise ValueError("--num_nodes allocates one Spark/HDFS master and n-1 workers, and thus must be greater "
-                         "than 1. %d was passed." % options.num_nodes)
-
-    inputs = {'numWorkers': options.num_nodes - 1,
-              'outDir':     options.output_directory,
-              'bamName':    options.input_file_name,
-              'knownSNPs':  options.known_SNPs,
-              'driverMemory': options.driver_memory,
-              'executorMemory': options.executor_memory,
-              'sudo': options.sudo,
-              'suffix': None,
-              'masterIP': options.master_ip}
-
-    Job.Runner.startToil(Job.wrapJobFn(static_adam_preprocessing_dag, inputs), options)
-
-if __name__=="__main__":
-    # FIXME: main() doesn't return anything
-    sys.exit(main(sys.argv[1:]))
+def main():
+
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(dest='command')
+    # Generate subparsers
+    subparsers.add_parser('generate-config', help='Generates an editable config in the current working directory.')
+    # Run subparser
+    parser_run = subparsers.add_parser('run', help='Runs the ADAM preprocessing pipeline')
+    parser_run.add_argument('--config', default='adam_preprocessing.config', type=str,
+                            help='Path to the (filled in) config file, generated with "generate-config". '
+                                 '\nDefault value: "%(default)s"')
+    parser_run.add_argument('--sample', help='The full s3 url of the input SAM or BAM file')
+    parser_run.add_argument('--output-dir', default=None,
+                            help='full path where final results will be output')
+    parser_run.add_argument('-s', '--suffix', default='',
```

Comment:
It is not clear to me what the actual final output file paths will look like.  What is the difference between `--suffix` here and `dir-suffix` in the config file?